product/tutorials/tutorial_add_fulfillment_places.md

```diff
@@ -0,0 +1,147 @@
+
+These methods are asynchronous because of downstream optimizations that support hundreds of concurrent updates per product, without sacrificing performance.
+
+You can find detailed information about managing catalog information in the [Retail API documentation](https://cloud.google.com/retail/docs/inventory-updates#inventory-update-methods)
+
+
+**Time to complete**: 
+<walkthrough-tutorial-duration duration="3.0"></walkthrough-tutorial-duration>
+
+## Before you begin
+
+To run Python code samples from this tutorial, you need to set up your virtual environment.
+
+To do that, run the following commands in a terminal:
+
+```bash
+python3 -m venv tutorial-env
+```
+
+```bash
+source tutorial-env/bin/activate
+```
+
+Next, install Google packages:
+
+```bash
+python3 -m pip install google
+```
+
+```bash
+python3 -m pip install google.cloud.retail
+```
+
+**Tip**: Click the copy button on the side of the code box to paste the command in the Cloud Shell terminal and run it.
+
+## Set the PROJECT_NUMBER environment variable
+
+As you are going to run the code samples in your own Cloud Project, you should specify the **project_id** as an environment variable, it will be used in every request to the Retail API.
+
+You can find the ```project_number``` in the **Home/Dashboard/Project Info card**.
+
+Set the environment variable with a following command:
+```bash
+export PROJECT_NUMBER=<YOUR_PROJECT_NUMBER>
+```
+
+## Add fulfillment places 
+
+To send the AddFulfillmentPlacesRequest you need to set the following fields:
+ - ```product``` - the product name which inventory information you wnd to update,
+ - ```type``` - the fulfillment type, you may set commonly used types (such as pickup in store and same day delivery) or custom types,
+ - ```place_ids[]``` - the store IDs for each of the fulfillment type,
+ - ```add_time``` - the time when the fulfillment updates are issued, used to prevent out-of-order updates on fulfillment information. If not provided, the internal system time will be used,
+ - ```allow_missing``` - if set to true, and the Product is not found, the fulfillment information will still be processed and retained for at most 1 day and processed once the Product is created. 
+
+Open **product/add_remove_fulfillment.py** and check the AddFulfillmentPlacesRequest.
+
+To add the fulfillment places, open terminal and run the command:
+
+```bash
+python product/add_remove_fulfillment.py
+```
+
+Check the responses in the Terminal. As you can see, the product was initially created without the fulfillment information, 
+then check the get_product() response, the fulfillment places 'store1', 'store2', 'store3' was added to the fulfillment type 'pickup-in-store'.
+
+## Send out-of-order add_fulfillment_places request
+
+The AddFulfillmentPlaces method allows the caller to specify an update time for when the request is issued.
+This update time is compared against the latest update time recorded for the relevant inventory fields,
```

Comment:
The Retail API compares the update time you've specified with the latest recorded for the relevant inventory fields. The update happens only if the specified update time value is greater than the latest update time value.